deepinv/physics/generator/blur.py

```diff
@@ -0,0 +1,673 @@
+        if sigma is None:
+            sigma = self.sigma
+        if l is None:
+            l = self.l
+        fraction = 5**0.5 * diff.abs() / l
+        return sigma**2 * (1 + fraction + fraction**2 / 3) * torch.exp(-fraction)
+
+    # @torch.compile
+    def f_matern(self, batch_size: int = 1, sigma: float = None, l: float = None):
+        vec = torch.randn(batch_size, self.n_steps, **self.factory_kwargs)
+        time = torch.linspace(-torch.pi, torch.pi, self.n_steps, **self.factory_kwargs)[
+            None
+        ]
+
+        kernel = self.matern_kernel(time, sigma, l)
+        kernel_fft = torch.fft.rfft(kernel)
+        vec_fft = torch.fft.rfft(vec)
+        return torch.fft.irfft(vec_fft * torch.sqrt(kernel_fft)).real[
+            :,
+            torch.arange(self.n_steps // (2 * torch.pi), **self.factory_kwargs).type(
+                torch.int
+            ),
+        ]
+
+    def step(self, batch_size: int = 1, sigma: float = None, l: float = None):
+        r"""
+        Generate a random motion blur PSF with parameters :math:`\sigma` and :math:`l`
+
+        :param int batch_size: batch_size.
+        :param float sigma: the standard deviation of the Gaussian Process
+        :param float l: the length scale of the trajectory
+
+        :return: dictionary with key **'filter'**: the generated PSF of shape `(batch_size, 1, psf_size[0], psf_size[1])`
+        :rtype: dict
+        """
+        ## add batch size to the shape. We can have a different batch size at each call of step()
+        ## We enforce only one channel as the underlying generator code only works for one channel at the time
+        # if self.shape[0] != 1:
+        #    self.shape = (batch_size, 1, self.shape[-2], self.shape[-1])
+        # else:
+        #    self.shape = (batch_size, self.shape[-3], self.shape[-2], self.shape[-1])
+
+        f_x = self.f_matern(batch_size, sigma, l)[..., None]
+        f_y = self.f_matern(batch_size, sigma, l)[..., None]
+
+        trajectories = torch.cat(
+            (
+                f_x - torch.mean(f_x, dim=1, keepdim=True),
+                f_y - torch.mean(f_y, dim=1, keepdim=True),
+            ),
+            dim=-1,
+        )
+        kernels = [
+            histogramdd(trajectory, bins=list(self.psf_size), low=[-1, -1], upp=[1, 1])[
+                None, None
+            ]
+            for trajectory in trajectories
+        ]
+        kernel = torch.cat(kernels, dim=0)
+        kernel = kernel / torch.sum(kernel, dim=(-2, -1), keepdim=True)
+
+        return {
+            "filter": kernel.expand(
+                batch_size,
+                self.num_channels,
+                self.psf_size[0],
+                self.psf_size[1],
+            )
+        }
+
+
+class DiffractionBlurGenerator(PSFGenerator):
+    r"""
+    Diffraction limited blur generator.
+
+    Generates 2D diffraction kernels in optics using Zernike decomposition of the phase mask (Fresnel/Fraunhoffer diffraction theory)
```

Comment:
```suggestion
    Generates 2D diffraction kernels in optics using Zernike decomposition of the phase mask (Fresnel/Fraunhoffer diffraction theory).
```

This is great! Maybe add a ref?